src/java.base/share/classes/jdk/internal/util/random/RandomSupport.java

```diff
@@ -1210,15 +1210,16 @@ public static double computeNextExponentialSoftCapped(RandomGenerator rng, doubl
             return DoubleZigguratTables.exponentialX[(int)i] * (U1 >>> 1);
         }
         // We didn't use the upper part of U1 after all.  We'll probably be able to use it later.
-        if (maxValue <= 0.0) {
-            return 0.0;
+        if (maxValue <= DoubleZigguratTables.exponentialX0) {
+            return DoubleZigguratTables.exponentialX0;
         }
         final long maxExtraMinus1;
-        if (maxValue >= MAX_EXPONENTIAL) {
+        if (maxValue >= Math.nextDown(MAX_EXPONENTIAL)) {
             maxExtraMinus1 = Long.MAX_VALUE;
         } else {
-            // Conversion to long rounds toward zero
-            maxExtraMinus1 = (long) (maxValue / DoubleZigguratTables.exponentialX0);
+            // Math.round rounds toward infinity in conversion to long; adding 1.0 corrects for any
+            // downward rounding error in the division
+            maxExtraMinus1 = Math.round(1.0 + maxValue / DoubleZigguratTables.exponentialX0);
```

Comment:
We had asked Guy Steele for his inputs on these proposed changes. Guy reviewed these changes and for this specific if/else block change, he notes that (I've paraphrased it):

> the (proposed new) computation of `maxExtraMinus1` looks okay to me: it’s always okay for maxExtraMinus1 to be a bit larger than you might expect; the only downside is that the (for) loop starting (on the next line) might take extra iterations.

Reply:
Is Guy saying it's impossible for `maxExtraMinus1` to be too small even without this change?